### ISSUE TITLE
If cspm is enabled, `the server could not find the requested resource` continues

### ISSUE THREAD

**Issue by inobu** [DataDog/datadog-operator#975]
**Output of the info page (if this is a bug)**
```
RBAC: allowed by ClusterRoleBinding "default-datadog-cspm-cluster-agent" of ClusterRole "default-datadog-cspm-cluster-agent" to ServiceAccount "datadog-cluster-agent/default"
```

**Describe what happened:**
If cspm is enabled, `the server could not find the requested resource` continues.
Therefore, at least in GKE, control plane upgrades are not automatically performed.

**Describe what you expected:**
There will be no errors

**Steps to reproduce the issue:**
```yaml
apiVersion: datadoghq.com/v2alpha1
kind: DatadogAgent
metadata:
  name: datadog
spec:
  features:
    cspm:
      enabled: true
```

**Additional environment details (Operating System, Cloud provider, etc):**
GKE 1.25

**Comment by levan-m** [comment]
Hello, thanks for reporting the issue! Could you please share Operator, chart versions and environment (local Kind, public cloud etc.) you are using?

**Comment by inobu** [comment]
@levan-m 
Hi!

my environment is here

chart 1.0.3
target revision 1.0.4
Public Cloud（GKE 1.25〕

**Comment by levan-m** [comment]
I tried chart 1.0.3 which installs Operator 1.0.0. I'm not sure what's the 1.0.4 version you are referring to. 

I can see similar line logged by cluster agent - not sure if that's what you are encountering. 
```
2023-11-09 21:11:50 UTC | CLUSTER | WARN | (pkg/compliance/rego/rego_check.go:307 in buildNormalInput) | failed to resolve input: unable to list Kube resources:'policy/v1beta1, Resource=podsecuritypolicies', ns:'' name:'', err: the server could not find the requested resource
```
File which logs above line was removed/refactored in 7.46 - https://github.com/DataDog/datadog-agent/pull/15758. I don't see that getting logged when installing agent with Operator chart 1.2.1 (which uses 7.47.1), can't really tell whether issue is still there but not logged.

Have you tried using more recent version of Operator/Agent? Anything blocking you from upgrade?
